libcrux-sha3/tests/sha3.rs

```diff
@@ -1,3 +1,343 @@
+                  c99fc734804563770d518c27aa3f9e2714d8e945b4df71d5c4d6b6d91e2f\
+                  981ff84e260e2011618bbd3d59ec07948eee3de448b8916d19fda8152f55\
+                  78108506cdb5b8103956dc80c789085c0af06483a9892e4b1ff0d97";
+        pub const FOX_53: &str =
+            "f4202e3c5852f9182a0430fd8144f0a74b95e7417ecae17db0f8cfeed0e3e66eb\
+                  5585ec6f86021cacf272c798bcf97d368b886b18f";
+    }
+
+    pub mod shake256 {
+        pub const HELLO_FIVE_BLOCKS: &str =
+            "b3be97bfd978833a65588ceae8a34cf59e95585af62063e6b89d0789f372424e8\
+                  b0d1be4f21b40ce5a83a438473271e0661854f02d431db74e6904d6c347d\
+                  757a33b44f18e740bd119782f48b0ac4ee1fa2dee4c5018ee2f186d0ff94\
+                  d1cece111e29a6bbd0972cb8574b5afddd55f00e50bd402c998043ba3f45\
+                  53558391be010abb209af935224b8c331d0d29c008185f2c900abad89885\
+                  1c4f3d941a13f03e3c315c4fb058fca2bb4e2bc53fec7866eb7e7636f276\
+                  dc5a167cad77b286c9a94946fe054927c48db7f30424787f56153cc67ca4\
+                  9609928d24c16563d3a0aaad1ca1495003374868ec422a72bedd2f387abc\
+                  350b46a9a6580a3ceb56b602b7edab836d58d8bb6b1a6975aaad42554132\
+                  71ec544ddea12dbb65003da4273650d6e3b51373e4e86fced975dad607ad\
+                  d1184702952d4bf8459d05197293d35b59688a9f13806887f9845211eb2d\
+                  0b9cc1e089eba8c16f9967d80ec181a754ea6511a897c736ba4c09871d99\
+                  3a41cf7efb08f0479935eaa811865002353f39594d432417d0e70d371509\
+                  bb0b76003e9712354427ab1e4f69ebd5e32b585166b3e843b062efa32bc7\
+                  1bbdc0989b87137752452a8a908ccea6ee1980e9213c6a380cdb947be228\
+                  5416b088ee4646793286d44b25df89575df2ef08a4c78237e7e25ec8b3a3\
+                  af7a63c0aa0fd46582874ab9417fb4e720298a4d6de8faa6f71a4ef4e6a1\
+                  4a5dcce0f002465987e661e9ed0d39fa79d018572ac40613630bf68868de\
+                  5cbe1e33eb014cdeeb125f8842fd1b0bd3c4970f2ddb9a3db5cdd0ca7e37\
+                  785d2029bbe2e6a8a225265fbbdd12e9712a538f5a346eeab6f9cc296580\
+                  e6d7c274d07084e758d01006b22bd45778ecb86bb495d413aef4dc28aa84\
+                  8f46cbe4e189fb0d3de54bf2c146d280b163e9358200547ee71207f11a4e\
+                  25e643a4552d6971cf4efb277a7d1d10095";
+
+        pub const FOX_71: &str =
+            "2f671343d9b2e1604dc9dcf0753e5fe15c7c64a0d283cbbf722d411a0e36f6ca1\
+                  d01d1369a23539cd80f7c054b6e5daf9c962cad5b8ed5bd11998b40d5734442bed798f6e5c915";
+    }
+}
+
+// Portable implementation tests
+mod portable {
+    use super::test_vectors;
+    use libcrux_sha3::portable::incremental::Xof;
+    use libcrux_sha3::portable::{incremental, sha224, sha256, sha384, sha512, shake128, shake256};
+
+    #[test]
+    fn sha3_224() {
+        let mut digest = [0u8; 28];
+
+        sha224(&mut digest, test_vectors::EMPTY);
+        assert_eq!(hex::encode(&digest), test_vectors::sha3_224::EMPTY);
```

Comment:
For all of these: `digest` is a reference already. So no need for the additional borrow.